stdpopsim/slim_engine.py

```diff
@@ -65,10 +66,14 @@
 
     defineConstant("generation_time", $generation_time);
     defineConstant("mutation_rate", Q * $mutation_rate);
-    defineConstant("recombination_rate", (1-(1-2*$recombination_rate)^Q)/2);
     defineConstant("chromosome_length", $chromosome_length);
     defineConstant("trees_file", "$trees_file");
     defineConstant("check_coalescence", $check_coalescence);
+
+    _recombination_rates = $recombination_rates;
+    _recombination_ends = $recombination_ends;
+    defineConstant("recombination_rates", (1-(1-2*_recombination_rates)^Q)/2);
```

Comment:
We're introducing some floating-point error here (since x != (1 - (1 - x))). I suppose this is harmless but a little annoying.

Reply:
I also don't see what the factor of 2 is doing?

Reply:
In ignorance, I copied this from the SLiM manual section 5.5 (p. 103 in the version I have), where one Peter Ralph is credited. :P

Reply:
> where one Peter Ralph is credited. :P
Hm, dubious. =)
> This formula is based upon the probability that a binomial draw will be odd
Oh, right.  Ok, all good.